src/libraries/System.Private.CoreLib/src/System/Threading/Wasi/WasiEventLoop.cs

```diff
@@ -178,7 +180,9 @@ public void ResolveAndDispose()
 
                 // no need to unregister the holder from s_pollables, when this is called
                 isDisposed = true;
-                pollable.Dispose();
+                if (ownsPollable){
```

Comment:
This is the interesting part.

Related: The new wit-bindgen doesn't generate finalizers anymore. That's good because resources can't be disposed in random order.
Related: when we take pollable handles from `libc` internals, they are still owned by `libc` and we should not dispose them here. That's why the new parameter, which will be used in sockets PR